ostelco-ios-client/Features/App/GlobalStore.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  GlobalStore.swift
+//  ostelco-ios-client
+//
+//  Created by mac on 10/25/19.
+//  Copyright © 2019 mac. All rights reserved.
+//
+
+import ostelco_core
+
+final class GlobalStore: ObservableObject {
+    @Published var country: Country?
+    @Published var previousCountry: Country?
+    @Published var regions: [PrimeGQL.RegionDetailsFragment]?
+    @Published var countryCodeToRegionCodeMap = [:] as [String: [String]]
+    
+    init() {
+        
+        country = LocationController.shared.currentCountry
+
+        if let countryCode = UserDefaultsWrapper.countryCode {
+            previousCountry = Country(countryCode)
+        }
+        
+        NotificationCenter.default.addObserver(self, selector: #selector(countryChanged(_:)), name: CurrentCountryChanged, object: nil)
+        
+        loadRegions()
+    }
+    
+    func loadRegions() {
+        APIManager.shared.primeAPI.loadRegions()
+        .done { self.regions = $0 }.cauterize()
+    }
+    
+    @objc func countryChanged(_ notification: NSNotification) {
+        guard let controller = notification.object as? LocationController else {
+            fatalError("Something other than the location controller is posting this notification!")
+        }
+        country = controller.currentCountry
+        UserDefaultsWrapper.countryCode = country?.countryCode
+    }
+    
+    func simProfilesForCountry(country: Country) -> [PrimeGQL.SimProfileFields] {
+        if let regionDetailsList = regions, let regionCodes = countryCodeToRegionCodeMap[country.countryCode] {
+            return regionDetailsList
+                .filter({ regionCodes.contains($0.region.id) })
+                .filter({ $0.simProfiles != nil })
+                .map({ $0.simProfiles!.map({ $0.fragments.simProfileFields }) })
+                .reduce([], +)
+        }
+        return []
+    }
+    
+    func showCountryChangedMessage() -> Country? {
+        if let previousCountry = previousCountry, let country = country {
```

Comment:
The store appears to be caching the value to decide if the country has changed.

Reply:
Since the store is responsible for updating the UI the changes has to go through the store. That also means the store has to listen for the country changes itself.
That's at least my current understanding of SwiftUI and view updates, if you have a better idea on how to solve this, I'm all ears

Reply:
I’m saying the logic to decide when the country is different should not be here. This should be able to be notified of the change and be done. Now the job of keeping track of countries has been spread between two entities.

Reply:
I'm not sure how you want this to be, it would probably be easier if you do it, to show me what you are thinking about